tests/system_tests/test_core/test_wandb_metadata.py

```diff
@@ -40,3 +35,12 @@ def test_metadata_ops(user, disabled: bool):
     assert run._metadata.gpu_count == 420
 
     run.finish()
+
+
+@pytest.mark.skip_wandb_core
+def test_metadata_legacy(user):
+    run = wandb.init()
+
+    # Legacy service does not support metadata
+    assert run._metadata is None
```

Comment:
should it be None or just no-op? if i turn legacy on for something all my scripts will start braking because i can't access any attributes on metadata

Reply:
ok, I wired it up with the legacy service, omitting the update part - that is a noop, so your run with legacy on is safe ;)